src/circuit.ts

```diff
@@ -49,12 +58,100 @@ export class Circuit {
+
+        const file = fs.createWriteStream(outputPath);
+        return new Promise((resolve, reject) => {
+            https.get(fileUrl, function(response) {
+                response.pipe(file);
+
+                file.on("finish", () => {
+                    file.close();
+                    resolve(true)
+                });
+            });
+        })
+    }
+
+    getTauFileUrlByConstraints (c: number):string {
+        let url = ""
+
+        const tauRangeMap ={
+            256: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_08.ptau",
+            512: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_09.ptau",
+            1000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_10.ptau",
+            2000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_11.ptau",
+            4000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_12.ptau",
+            8000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_13.ptau",
+            16000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_14.ptau",
+            32000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_15.ptau",
+            64000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_16.ptau",
+            128000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_17.ptau",
+            256000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_18.ptau",
+            512000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_19.ptau",
+            1000000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_20.ptau",
+            2000000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_21.ptau",
+            4000000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_22.ptau",
+            8000000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_23.ptau",
+            16000000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_24.ptau",
+            32000000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_25.ptau",
+            64000000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_26.ptau",
+            128000000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_27.ptau",
+            256000000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final.ptau",
+        }
+
+        for (var key in tauRangeMap) {
+            if (Object.prototype.hasOwnProperty.call(tauRangeMap, key)) {
+                if(c <= (parseInt(key))) {
+                    url = tauRangeMap[key]
+                    break;
+                }
+            }
+        }
+
+        return url
+    }
+ 
+    async downloadPowerOfTauFile(){
+        const tauFolderPath =  this._circuitConfig.outputDir + '/tau'
```

Comment:
The tauFolderPath should be `path.resolve(this._circuitConfig.outputDir + '../tau')` and not  `path.resolve(this._circuitConfig.outputDir + './tau')`.

Otherwise there is no common folder these circuits are sharing!